nmi_mysql/nmi_mysql.py

```diff
@@ -11,7 +11,31 @@
 
 class DB(object):
 
-    def __init__(self, conf, autoconnect=False):
+    def __init__(self, conf, max_pool_size=20):
+        self.conf = conf
+        self.max_pool_size = max_pool_size
+        self._initialize_pool()
+
+    def _initialize_pool(self):
+        self.pool = Queue(maxsize=self.max_pool_size)
+
+        for _ in range(0, self.max_pool_size):
+            self.pool.put_nowait(Connection(self.conf))
+
+    def connect(self):
+        con = self.pool.get(True)
+        con.connect()
+
+        return con
+
+    def close(self, con):
+        con.close()
+        self.pool.put_nowait(con)
```

Comment:
In most connection pool, when a connection it is closed, you won't be able to reuse it (a new one need to be created), now I'm not sure if it is the same for pymysql.

But I guess since you require user to call connect themselves every time (instead of just get a pooled connection), this is fine.


A general guide of a pool connection:
1. user creates a connection pool
2. user just keep getting connection from the pool
2-1. user would be blocked if connection pool has no more connections
2-2. connection is returned to the pool when a query is finished (automatically)
3. user calls pool.close() when they don't need the connection anymore.

With this workflow the user never need to close/re-open the connection, otherwise, it defeats the purpose of connection pooling because each connection is re-established.